packages/strapi-design-system/src/NumberInput/NumberInput.js

```diff
@@ -124,10 +146,29 @@ export const NumberInput = React.forwardRef(
             size={size}
             endAction={
               <>
-                <ArrowButton aria-hidden reverse onClick={increment} tabIndex={-1} type="button">
+                <ArrowButton
+                  aria-hidden
+                  reverse
+                  onClick={() => {
+                    // increment needs an argument, so we can't remove the parenthesis
+                    increment();
+                  }}
+                  tabIndex={-1}
+                  type="button"
+                  data-testid="ArrowUp"
```

Comment:
Don't we need to add the disabled props here? To fix the disabled state

Reply:
you're right good catch, on it!